crates/provider-wit-bindgen/tests/bindgen_kv_memory.rs

```diff
@@ -0,0 +1,113 @@
+use std::process::Stdio;
+
+use anyhow::{Context, Result};
+use escargot::CargoBuild;
+use tempfile::TempDir;
+use tokio::process::Command;
+use uuid::Uuid;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_kv_memory() -> Result<()> {
+    let wasmcloud_bin_path =
+        std::env::var("TEST_WASMCLOUD_BIN").context("no wasmcloud bin provided")?;
+    let wash_binary_path = std::env::var("TEST_WASH_BINARY").unwrap_or("wash".into());
```

Comment:
We should not be depending on `wash`, we already have the code to build providers https://github.com/wasmCloud/wasmCloud/blob/main/tests/providers/build.rs, why reinvent the wheel?

Reply:
`wash` is how end users use wasmcloud -- it should be covered under test.

Note that the code that is in use here is using *what users would use* -- it's an E2E test.

The bindgen work is an *external* facing tool -- when people use it to make providers, they're going to use `wash` to create par files if they so choose, and run the wasmcloud binary. I want to cover that situation under test and make sure it does not regress.

Reply:
This does not belong here, using `wash` is out-of-scope for this repository. This project is a *dependency* of `wash` and we do not need recursive dependency cycles